### PR TITLE
Allow user  to set user-defined blob metadata for blob

### PR DESCRIPTION
**Pull request by ArchangelSDY**
This PR implements feature request #8 .

### CODE REVIEW

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 1]
```diff
@@ -338,7 +335,19 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
...
-				blobProperties.configure(blob);
+				blobProperties.configure(blob, env);
+			}
+
+			// Set blob metadata
+			if (metadata != null) {
+				HashMap<String, String> metadataMap = blob.getMetadata();
+				for (AzureBlobMetadataPair pair : metadata) {
+					metadataMap.put(
+						Util.replaceMacro(pair.getKey(), env),
```
Comment (chenkennt) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: what if key is empty?

Comment (ArchangelSDY) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: Looks like Azure SDK will throw IllegalArugmentException for null, empty and white space metadata key/value. I've added some check.

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 2]
```diff
@@ -342,10 +343,22 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
...
 				for (AzureBlobMetadataPair pair : metadata) {
-					metadataMap.put(
-						Util.replaceMacro(pair.getKey(), env),
-						Util.replaceMacro(pair.getValue(), env)
-					);
+					final String resolvedKey = Util.replaceMacro(pair.getKey(), env);
+					final String resolvedValue = Util.replaceMacro(pair.getValue(), env);
+
+					// Azure does not allow null, empty or whitespace metadata key
+					if (resolvedKey == null || resolvedKey.trim().length() == 0) {
```
Comment (chenkennt) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: Seems there is already a `Utils.isNullOrEmpty()` method to check whether a string is null or empty. There're also some places that use `StringUtils.isBlank()`. Can we consolidate them to use the same method?

Comment (chenkennt) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: This hasn't changed.

Comment (ArchangelSDY) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java]: Fixed. Looks like `Utils.isNullOrEmpty` does nothing more than `StringUtils.isBlank` so I removed it.

**Pull request merged**